### PR TITLE
Remove Lightning Kokkos plugin

### PR DESCRIPTION
**Pull request by rauletorresc**
**Context:** Lightning Kokkos plugin has been migrated to Lightning. We can remove its code from Catalyst then. But this PR in Lightning must be approved first: https://github.com/PennyLaneAI/pennylane-lightning/pull/837

**Description of the Change:** Remove plugin from code and from the list of provided plugins.

**Benefits:** Remove duplicated functionality.

[sc-69450]

### CODE REVIEW

**Comment by rauletorresc** [comment]
Is this correct?

**Comment by dime10** [comment]
> Is this approach correct?

That looks right to me. The next step is updating the lightning-kokkos dependency to a recent dev wheel and checking whether we can run all our tests.

Actually, I'm wondering whether the PR that moved the plugin also ensured the plugin was distributed with the their wheels. I guess we will find out 😅

**Comment by dime10** [comment]
Removing the wheel build label until we've finished repairing the actions.

File: doc/dev/installation.rst
Comment (dime10) [inline doc/dev/installation.rst:353]: We might want to think about what happens with the runtime tests Ali wrote for the device. Were they migrated over? Or are we just dropping them? @maliasadi

**Review by dime10 — APPROVED** [review]
Love to see it! Let's wait until the lightning PR is merged and also until we can test all wheel configurations, but this has my preliminary approval :)

**Comment by dime10** [comment]
Just waiting on the lightning PR to finish testing these changes 👍

File: doc/dev/installation.rst
Comment (maliasadi) [reply doc/dev/installation.rst:353]: Yes, they are all migrated to the Lightning repo and we can drop them all in the runtime.

File: .dep-versions
Comment (dime10) [inline .dep-versions:16]: Updating the wheel actions is missing right?

Comment (dime10) [reply .dep-versions:16]:
```suggestion
# For a custom LQ/LK version: update the package version here and at
# 'doc/requirements.txt'. Also, update the 'LIGHTNING_GIT_TAG' at 
# 'runtime/Makefile' with the exact commit hash corresponding to
# the merged PR that implements the desired feature.
```

**Pull request merged**
